etcdserver/server.go

```diff
@@ -222,6 +222,16 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		return nil, err
 	}
 
+	isSuccess := false
+	defer func() {
+		if !isSuccess {
+			// It removes member directory when NewServer returns error.
+			// This prevents conflicts with 'proxy' directory when
+			// the node falls back to proxy.
+			os.RemoveAll(cfg.MemberDir())
```

Comment:
I feel this is pretty dangerous thing to do. If user has a broken wal and etcd cannot start, this will remove all user data.

What we really need to do is not to create member dir when discovery fails.